File: src/main/java/art/arcane/racbot/util/ColorParser.java
package art.arcane.racbot.util;

import java.awt.Color;

public final class ColorParser {
  private ColorParser() {}

  public static Color parseHex(String color) {
    String normalized = color == null ? "" : color.trim();
    if (normalized.startsWith("#")) {
      normalized = normalized.substring(1);
    }
    return new Color(Integer.parseInt(normalized, 16));
  }
}

File: lib/src/util/color_parser.dart
import 'package:nyxx/nyxx.dart';

class ColorParser {
  const ColorParser._();

  static DiscordColor parseHex(String? color) {
    String normalized = color == null ? '' : color.trim();
    if (normalized.startsWith('#')) {
      normalized = normalized.substring(1);
    }
    return DiscordColor(int.parse(normalized, radix: 16));
  }
}
